metricbeat/helper/module.go

```diff
@@ -227,6 +244,7 @@ func (m *Module) processEvents(events []common.MapStr, metricSet *MetricSet) ([]
 			eventFieldName: event,
 			"metricset":    metricSet.Name,
 			"module":       m.name,
+			"fields":       m.Config.Fields,
```

Comment:
Publish this as `common.EventMetadatKey: m.Config.EventMetadata,`. The output will look for that key in the event and then properly merge the fields and tags with the "global" fields and tags.